blockstore_test.go

```diff
@@ -143,9 +143,11 @@ func TestAllKeysRespectsContext(t *testing.T) {
 
 	cancel()
 
-	v, ok = <-ch
-	require.Equal(t, cid.Undef, v)
-	require.False(t, ok)
+	received := 0
+	for range ch {
+		received++
+		require.LessOrEqual(t, received, 1, "expected to receive at most one result after the context was canceled")
```

Comment:
Thats a very strange set of semantics to expect here. I guess it makes sense, but is this really what we want the blockstore interface to guarantee?

Reply:
No, we don't really have _any_ guarantees. The blockstore is allowed to buffer responses.

Reply:
I.e., it could give us 10 items. I guess I can just make the test allow that to be more flexible?

Reply:
Certainly not specified by the spec, but I assume this is only testing one blockstore implementation, so you can make more assumptions :)